homeassistant/components/unifiled/config_flow.py

```diff
@@ -0,0 +1,64 @@
+"""Config flow for unifiled integration."""
+import logging
+
+from unifiled import unifiled
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({"host": str, "port": str, "username": str, "password": str})
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+
+    try:
+        unifiled(
+            data["host"],
+            data["port"],
+            username=data["username"],
+            password=data["password"],
+        )
+        return {"title": data["host"]}
+    except Exception as ex:  # pylint: disable=broad-except
```

Comment:
We want to catch a more specific error first. We can fall back to broad exception catching if needed.

Reply:
Changed